enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,428 @@
+
+### Use Case #1 - Bare Metal, Primary Network VLAN
+
+Consider an on-premise environment where the network administrator has isolated
+their provisioning infrastructure on a separate network from where the cluster
+is expected to operate.  It would be wasteful to dedicate an entire NIC just
+for provisioning and no other purpose.  Instead, the administrator would like
+to bring up the cluster’s primary network as a VLAN on the same network
+interface that was used for provisioning.
+
+* Bare metal servers with a single network interface
+* We desire PXE booting these servers on the untagged network for provisioning
+* We must configure a VLAN on this interface to access the primary external
+  network for the cluster
+* Access to this network must be configured very early, before ignition will be
+  able to reach the machine config server
+
+### Use Case #2 - Extra VLAN, Post Install Configuration
+
+Consider an on-premise environment with workloads that must be able to access
+resources on multiple networks that must be directly connected to the cluster.
+One example could be an isolated network running database services that are not
+routeable from any other network.  Instead, we’ll attach the Nodes to that VLAN
+and limit which pods can access it.
+
+* After cluster installation, configure another VLAN on a network interface
+  that specific workloads will need access to
+
+### Use Case #3 - Static Network Configuration Only
+
+Consider a lab environment where DHCP services either aren’t available at all,
+or the person trying out an OpenShift cluster does not have access to it to
+configure DHCP address reservations.  As long as the cluster administrator has
+a pool of addresses they are allowed to assign manually, a cluster with fully
+static network configuration would allow them to install OpenShift without
+needing to work with the network administrator on DHCP server configuration.
+
+* Must be able to override default RHCOS behavior of requiring DHCP to be
+  present on at least one interface
+* Must be able to configure static network configuration for every interface
+  used by the cluster
+
+
+## Existing Configuration Interfaces
+
+### Summary
+
+First, here’s a summary of how an administrator may configure host networking
+in a cluster.  More details about each interface are in the following sections.
+
+#### First Boot
+
+When an RHCOS host first boots, the initramfs environment must be able to reach
+the [Machine Config
+Server](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigServer.md)
+for Ignition to succeed.  Ignition will attempt to reach the Machine Config
+Server via an included URL. By default, dracut will attempt to do DHCPv4 and
+DHCPv6 on all interfaces and wait for one of them to work.  To change this
+behavior, an admin can:
+
+* Change the `ip=<...>` kernel argument(s) via PXE configuration or at the grub
+  command line during boot
```

Comment:
Note that in https://github.com/coreos/ignition/issues/979 it's been proposed that virt-customizing the RHCOS image to modify the arguments in the grub config should be supported, although I'm not clear if we've got any docs mentioning that of if it's just a short-term workaround.

Reply:
Hm, I suppose that works for UPI installs.  For all the IPI installs, there's not really a convenient place to do this.  I'll note it.

Reply:
i made a more general comment that this is possible on bare metal